ddl/job_table.go

```diff
@@ -309,6 +314,118 @@ func (d *ddl) markJobProcessing(sess *session, job *model.Job) error {
+				time.Sleep(time.Second)
+				continue
+			}
+		case <-d.ctx.Done():
+			return
+		}
+		d.loadBackfillJobAndRun()
+	}
+}
+
+func (d *ddl) getTableByTxn(store kv.Storage, schemaID, tableID int64) (*model.DBInfo, table.Table, error) {
+	var tbl table.Table
+	var dbInfo *model.DBInfo
+	err := kv.RunInNewTxn(d.ctx, store, false, func(ctx context.Context, txn kv.Transaction) error {
+		t := meta.NewMeta(txn)
+		var err1 error
+		dbInfo, err1 = t.GetDatabase(schemaID)
+		if err1 != nil {
+			return errors.Trace(err1)
+		}
+		tblInfo, err1 := getTableInfo(t, tableID, schemaID)
+		if err1 != nil {
+			return errors.Trace(err1)
+		}
+		tbl, err1 = getTable(store, schemaID, tblInfo)
+		return errors.Trace(err1)
+	})
+	return dbInfo, tbl, err
+}
+
+func (d *ddl) loadBackfillJobAndRun() {
+	isDistReorg := variable.DDLEnableDistributeReorg.Load()
+	if !isDistReorg {
+		return
+	}
+	se, err := d.sessPool.get()
+	defer d.sessPool.put(se)
+	if err != nil {
+		logutil.BgLogger().Fatal("dispatch backfill jobs loop get session failed, it should not happen, please try restart TiDB", zap.Error(err))
+	}
+	sess := newSession(se)
+
+	runningJobIDs := d.backfillCtxJobIDs()
+	if len(runningJobIDs) >= reorgWorkerCnt {
+		return
+	}
+
+	// TODO: Add ele info to distinguish backfill jobs.
+	// Get a Backfill job to get the reorg info like element info, schema ID and so on.
+	bfJob, err := GetBackfillJobForOneEle(sess, runningJobIDs, InstanceLease)
+	if err != nil || bfJob == nil {
+		if err != nil {
+			logutil.BgLogger().Warn("[ddl] get backfill jobs failed in this instance", zap.Error(err))
+		} else {
+			logutil.BgLogger().Debug("[ddl] get no backfill job in this instance")
+		}
+		return
+	}
+
+	jobCtx, existent := d.setBackfillCtxJobContext(bfJob.JobID, bfJob.Meta.Query, bfJob.Meta.Type)
+	if existent {
+		logutil.BgLogger().Warn("[ddl] get the type of backfill job is running in this instance", zap.String("backfill job", bfJob.AbbrStr()))
+		return
+	}
+	// TODO: Adjust how the non-owner uses ReorgCtx.
+	d.setReorgCtxForBackfill(bfJob)
+	d.wg.Run(func() {
+		defer func() {
+			d.removeBackfillCtxJobCtx(bfJob.JobID)
+			tidbutil.Recover(metrics.LabelBackfillWorker, "runBackfillJobs", nil, false)
+		}()
+
+		if bfJob.Meta.ReorgTp == model.ReorgTypeLitMerge {
+			if !ingest.LitInitialized {
+				logutil.BgLogger().Warn("[ddl] we can't do ingest in this instance",
+					zap.Bool("LitInitialized", ingest.LitInitialized), zap.String("bfJob", bfJob.AbbrStr()))
+				return
+			}
+			logutil.BgLogger().Info("[ddl] run backfill jobs with ingest in this instance", zap.String("bfJob", bfJob.AbbrStr()))
+			err = runBackfillJobsWithLightning(d, sess, bfJob, jobCtx)
+		} else {
+			logutil.BgLogger().Info("[ddl] run backfill jobs with txn-merge in this instance", zap.String("bfJob", bfJob.AbbrStr()))
+			_, err = runBackfillJobs(d, nil, sess, bfJob, jobCtx)
+		}
+
+		if err == nil {
+			err = syncBackfillHistoryJobs(sess, d.uuid, bfJob)
```

Comment:
When the session `sess` is used, it may be concurrently used by another goroutine because it is put to the session pool already.